.github: Add ec2 information as a step

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63784

Also creates the common.yml.j2 file as a place to store common code
amongst the templates

Should look like: 
![image](https://user-images.githubusercontent.com/1700823/130495226-f18b8c0f-1ea7-4097-8bbb-e998fabb71f2.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30490682](https://our.internmc.facebook.com/intern/diff/D30490682)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra

seemethere committed:
.github: Add ec2 information as a step

Also creates the common.yml.j2 file as a place to store common code
amongst the templates

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add ec2 information as a step"


Also creates the common.yml.j2 file as a place to store common code
amongst the templates

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30490682).

driazati commented on .github/scripts/display_ec2_information.sh:
```diff
@@ -0,0 +1,12 @@
+
+function get_ec2_metadata() {
+    category=$1
+    curl -fsSL "http://169.254.169.254/latest/meta-data/${category}"
```
Can you add a comment about where this is hosted?

seemethere replied:
Done

seemethere committed:
Update on ".github: Add ec2 information as a step"


Also creates the common.yml.j2 file as a place to store common code
amongst the templates

Should look like: 
![image](https://user-images.githubusercontent.com/1700823/130495226-f18b8c0f-1ea7-4097-8bbb-e998fabb71f2.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30490682](https://our.internmc.facebook.com/intern/diff/D30490682)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30490682).
